### PR TITLE
Improve margin pool

### PR DESCRIPTION
**Pull request by haythem96**
# Improve margin pool

## High Level Description

This PR:
- Add SafeERC20 library
- Remove require statement on 0 amount transfer from pool to user.

### Code

- [x] Unit test 100% coverage
- [x] Does your code follow the naming and code documentation guidelines?

### Documentation

- [x] Is your code up to date with the spec? 
- [x] Have you added your tests to the testing doc?

### CODE REVIEW

File: test/controller.test.ts [hunk 1]
```diff
@@ -671,29 +671,6 @@ contract('Controller', ([owner, accountOwner1, accountOwner2, accountOperator1,
         )
       })
 
-      it('should revert withdrawing long otoken amount equal to zero', async () => {
-        const vaultCounter = new BigNumber(await controllerProxy.getAccountVaultCounter(accountOwner1))
-        assert.isAbove(vaultCounter.toNumber(), 0, 'Account owner have no vault')
```
Comment (aparnakr) [inline test/controller.test.ts:676]: wait can we still have this check in the controller?

**Review by aparnakr — APPROVED** [review]
LGTM, added a couple of comments

File: test/controller.test.ts (hunk 1 above)
Comment (haythem96) [reply test/controller.test.ts:676]: This check is totally unneeded, will just consume more gas for the user that is withdrawing more than zero amount.

**Review by antoncoding — DISMISSED** [review]
Haven't see you applied the SafeERC20 library to the margin pool, is this intentionally? 

From [Library](https://docs.openzeppelin.com/contracts/3.x/api/token/erc20)
To use this library you can add a using SafeERC20 for IERC20; statement to your contract, which allows you to call the safe operations as token.safeTransfer(…​), etc.

I think we can simply include that in the same PR!

**Review by antoncoding — APPROVED** [review]
I saw the thread on Slack. So this is just for removing the constraint, looks good to me

**Pull request merged**